### PR TITLE
Валиуллин - Лабораторная работа #2

### PR DESCRIPTION
**Pull request by DamirV**

### CODE REVIEW

**Comment by DamirV** [comment]
@VadimKutovoi 
@jTagWindWay

File: modules/radix/test/test_radix.cpp [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+TEST(RadixTest, Can_Create_Radix_With_Size) {
+    // Arrange
+    Radix rdx(20);
+
+    // Act & Assert
+    EXPECT_EQ(20, rdx.getSize());
+}
+
+TEST(RadixTest, Cant_Create_Radix_With_Size_EQ_Zero) {
```
Comment (jTagWindWay) [inline modules/radix/test/test_radix.cpp]: May be Throw_Then_Create_Radix_Zero_Size better

File: modules/radix/test/test_radix.cpp [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    // Arrange & Act & Assert
+    ASSERT_ANY_THROW(Radix rdx(0));
+}
+
+TEST(RadixTest, Cant_Create_Radix_With_Size_Less_Than_Zero) {
+    // Arrange & Act & Assert
```
Comment (jTagWindWay) [inline modules/radix/test/test_radix.cpp]: As well as 23# string. Please, check all your test case functions name.

File: modules/radix/test/test_radix.cpp [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    // Act & Assert
+    EXPECT_EQ(20, rdx.getSize());
+}
+
+TEST(RadixTest, Throw_Then_Create_Radix_Zero_Size) {
+    // Arrange & Act & Assert
+    ASSERT_ANY_THROW(Radix rdx(0));
+}
+
+TEST(RadixTest, Throw_Then_Create_Radix_With_Size_Less_Than_Zero) {
```
Comment (VadimKutovoi) [inline modules/radix/test/test_radix.cpp]: I think you mean "Throw_When", not "Throw_Then" :)

File: modules/radix/test/test_radix.cpp [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    // Arrange & Act & Assert
+    ASSERT_ANY_THROW(Radix rdx(-1));
+}
+
+TEST(RadixTest, Throw_Then_Create_Radix_With_Array_And_Size_Less_Than_Zero) {
```
Comment (VadimKutovoi) [inline modules/radix/test/test_radix.cpp]: Same, "When"

File: modules/radix/test/test_radix.cpp [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    // Arrange
+    int mas[] = { 1, 2, 3 };
+
+    // Act & Assert
+    ASSERT_ANY_THROW(Radix rdx(mas, -1));
+}
+
+TEST(RadixTest, Throw_Then_Create_Radix_With_Array_And_Zero_Size) {
```
Comment (VadimKutovoi) [inline modules/radix/test/test_radix.cpp]: Same

**Comment by DamirV** [comment]
@VadimKutovoi
@jTagWindWay

**Comment by jTagWindWay** [comment]
:+1:

**Comment by DamirV** [comment]
@TheG1uy

File: modules/radix/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# Declare variables for binaries' names
+get_filename_component(DIR_NAME ${CMAKE_CURRENT_LIST_DIR} NAME)
+set(MODULE      "${DIR_NAME}")
+set(LIBRARY     "lib_${MODULE}")
+set(TESTS       "test_${MODULE}")
+set(APPLICATION "app_${MODULE}")
+
+# Include directory with public headers
+include_directories("./")
```
Comment (alexbatashev) [inline modules/radix/CMakeLists.txt]: CMAKE_CURRENT_SOURCE_DIR

File: modules/radix/test/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,27 @@
+set(target ${TESTS})
+
+file(GLOB srcs "*.cpp")
+set_source_files_properties(${srcs} PROPERTIES
+    LABELS "${MODULE};Test")
+
+# VS2012 doesn't support correctly the tuples yet,
+# see http://code.google.com/p/googletest/issues/detail?id=412
+if(MSVC)
```
Comment (alexbatashev) [inline modules/radix/test/CMakeLists.txt]: Do you really need this?

**Comment by DamirV** [comment]
mistaken with the name of the commit, the corrected name "Added several tests, changed files CMakeLists.txt in folders radix/ and radix/test/"

**Comment by DamirV** [comment]
@alexbatashev 
check please

**Comment by alexbatashev** [comment]
@DamirV Please change the title according to the rules

**Comment by DamirV** [comment]
@alexbatashev The title of pull request?

**Pull request merged**